### DISCUSSION TITLE
Where can I find the hardware drivers

### DISCUSSION THREAD

**Discussion by aalbaali** [iRobotEducation/create3_docs#598] [Questions]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Ubuntu 22.04

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.6

### Which RMW is your robot running?

I don't know

### Is the robot connected to a network? If so, what is the network type?

Yes, home network

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

None

### Describe your question.

I'm experimenting with different middlewares and I'm trying to run the iCreate3 without using ROS (2), but I can't find the hardware drivers (e.g., encoders, motors, LEDs, etc.) nor their documentation.

What is the best way to control iCreate3 in a ROS-agnostic way? And, where can I find the documentation for these drivers?

**Comment by shamlian** [comment]
The iRobot Create 3 documentation is found [here](https://iroboteducation.github.io/create3_docs/). The options for controlling the robot are either ROS 2 over an Ethernet network, which is documented in detail [on the documentation site](https://iroboteducation.github.io/create3_docs/api/ros2/) or using the [iRobot Python SDK](https://github.com/iRobotEducation/irobot-edu-python-sdk)  which implements [this open protocol](https://github.com/iRobotEducation/root-robot-ble-protocol) over BLE.
